### PR TITLE
Fixes #1649 Add Proto Models for RatioExpression Interaction

### LINKED ISSUE #1649: Add new interaction objects for ratio expression
[Add new protos for the interaction objects needed for the Ratio Expression Input operation.

### PR DESCRIPTION
**Pull request by aggarwalpulkit596**
## Explanation

Fixes #1649 
Adds a new proto-model for the ratio interaction which works on the basis of Unicode with string validation for ratio(eg 1:2:3). For any clarification - [design doc](https://docs.google.com/document/d/1DfGt5VnfD1ueMfbLMq1J52tLYuYvRD6CemFop5_HtIA/edit#).

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by rt4914** [comment]
@aggarwalpulkit596 Please add description to issue or the PR because at this stage I have no idea how the Ratio interaction will work.

**Comment by aggarwalpulkit596** [comment]
@rt4914 updated. Let me know if this is not sufficient

File: model/src/main/proto/interaction_object.proto [hunk 1]
```diff
@@ -30,6 +31,11 @@ message StringList {
   repeated string html = 1;
 }
 
+// Structure containing a ratio object.
+message RatioExpression {
+  string html = 1;
```
Comment (BenHenning) [inline model/src/main/proto/interaction_object.proto]: Why wouldn’t we use a structured representation here like we would in the backend? We shouldn’t just store html that needs parsing. The data in there protos should be already structured (eg we should be able to easily observe the number of ratios, the components of the ratio, etc).

**Review by BenHenning — CHANGES_REQUESTED** [review]

File: model/src/main/proto/interaction_object.proto (hunk 1 above)
Comment (aggarwalpulkit596) [reply model/src/main/proto/interaction_object.proto]: I'm not sure if I'm following this correctly but does this mean we don't want to create a object of ratio explicitly in Android and can use the existing one's ?

Comment (aggarwalpulkit596) [reply model/src/main/proto/interaction_object.proto]: @BenHenning updated

File: model/src/main/proto/interaction_object.proto [hunk 2]
```diff
@@ -30,6 +31,11 @@ message StringList {
   repeated string html = 1;
 }
 
+// Structure containing a ratio object.
```
Comment (BenHenning) [inline model/src/main/proto/interaction_object.proto]: Provide an example here.

File: model/src/main/proto/interaction_object.proto [hunk 3]
```diff
@@ -30,6 +31,11 @@ message StringList {
...
+message RatioExpression {
+  repeated uint32 ratio_component = 1;
```
Comment (BenHenning) [inline model/src/main/proto/interaction_object.proto:39]: Suggest mentioning that it's expected that this list be more 1 element. 0 and 1 elements are invalid states.

**Review by BenHenning — CHANGES_REQUESTED** [review]
Thanks! Generally LGTM, but have a couple of nits to clarify the documentation around the new proto.

**Comment by BenHenning** [comment]
@aggarwalpulkit596 my comments don’t seem resolved, and I don’t see any open questions. Am I missing something?

File: model/src/main/proto/interaction_object.proto (hunk 3 above)
Comment (aggarwalpulkit596) [reply model/src/main/proto/interaction_object.proto:39]: Updated

**Comment by aggarwalpulkit596** [comment]
@BenHenning probably bug some GitHub issue. You can see it now i guess?

File: model/src/main/proto/interaction_object.proto (hunk 2 above)
Comment (aggarwalpulkit596) [reply model/src/main/proto/interaction_object.proto]: Done

File: model/src/main/proto/interaction_object.proto (hunk 3 above)
Comment (aggarwalpulkit596) [reply model/src/main/proto/interaction_object.proto:39]: Dond

File: model/src/main/proto/interaction_object.proto [hunk 4]
```diff
@@ -31,8 +31,10 @@ message StringList {
   repeated string html = 1;
 }
 
-// Structure containing a ratio object.
+// Structure containing a ratio object for eg - [1,2,3].
 message RatioExpression {
+  // It's also expected that list should have more than 1 elements with
```
Comment (BenHenning) [inline model/src/main/proto/interaction_object.proto:37]: ‘List of components in a ratio.’ + what you have here.

File: model/src/main/proto/interaction_object.proto [hunk 5]
```diff
@@ -31,8 +31,10 @@ message StringList {
   repeated string html = 1;
 }
 
-// Structure containing a ratio object.
+// Structure containing a ratio object for eg - [1,2,3].
```
Comment (BenHenning) [inline model/src/main/proto/interaction_object.proto]:
```suggestion
// Structure containing a ratio object for eg - [1,2,3] for 1:2:3.
```

**Review by BenHenning — APPROVED** [review]
Thanks @aggarwalpulkit596

File: model/src/main/proto/interaction_object.proto (hunk 4 above)
Comment (aggarwalpulkit596) [reply model/src/main/proto/interaction_object.proto:37]: done

File: model/src/main/proto/interaction_object.proto (hunk 5 above)
Comment (aggarwalpulkit596) [reply model/src/main/proto/interaction_object.proto]: done

**Pull request merged**